essentials-tp/src/main/kotlin/io/github/nyliummc/essentials/commands/CommandHome.kt

```diff
@@ -0,0 +1,175 @@
+                    onClickCommand("/home ${it.key}")
+                    color(Formatting.YELLOW)
+                    onHoverText("/home ${it.key}")
+                }
+            }
+        }
+        context.source.player.addChatMessage(text, false)
+        */
+        return 1
+    }
+
+    fun executeSet(context: CommandContext<ServerCommandSource>): Int {
+        val player = context.source.player
+        if (HomeCache.newHome(
+                        StoredHome(
+                                player.gameProfile,
+                                "home",
+                                Vec3i(player.posVector.x, player.posVector.y, player.posVector.z),
+                                player.dimension.rawId
+                        )
+                )) {
+            // player.addChatMessage(ConfigText(EssentialsModKt.messageConfig.tp.sethome.success, "home"), false)
+            return 1
+        }
+        // player.addChatMessage(ConfigText(EssentialsModKt.messageConfig.tp.sethome.failure, "home"), false)
+        return -1
+    }
+
+    fun executeSetTarget(context: CommandContext<ServerCommandSource>): Int {
+        val home = StringArgumentType.getString(context, "home")
+        val player = context.source.player
+        if (HomeCache.newHome(
+                        StoredHome(
+                                player.gameProfile,
+                                home,
+                                Vec3i(player.posVector.x, player.posVector.y, player.posVector.z),
+                                player.dimension.rawId
+                        )
+                )) {
+            // player.addChatMessage(ConfigText(EssentialsModKt.messageConfig.tp.sethome.success, home), false)
+            return 1
+        }
+        // player.addChatMessage(ConfigText(EssentialsModKt.messageConfig.tp.sethome.failure, home), false)
+        return -1
+    }
+
+    fun executeDel(context: CommandContext<ServerCommandSource>): Int {
+        val home = "home"
+        if (HomeCache.delHome(context.source.player.gameProfile, home)) {
+            // context.source.player.addChatMessage(ConfigText(EssentialsModKt.messageConfig.tp.delhome.success, home), false)
+            return 1
+        }
+        // context.source.player.addChatMessage(ConfigText(EssentialsModKt.messageConfig.tp.delhome.failure, home), false)
+        return -1
+    }
+
+    fun executeDelTarget(context: CommandContext<ServerCommandSource>): Int {
+        val home = StringArgumentType.getString(context, "home")
+        if (HomeCache.delHome(context.source.player.gameProfile, home)) {
+            // context.source.player.addChatMessage(ConfigText(EssentialsModKt.messageConfig.tp.delhome.success, home), false)
+            return 1
+        }
+        // context.source.player.addChatMessage(ConfigText(EssentialsModKt.messageConfig.tp.delhome.failure, home), false)
```

Comment:
Why not throw a CommandSyntaxException with your error message, this tells the game it actually failed unlike returning -1